cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -148,11 +148,15 @@ size_t batched_decompress_temp_size(compression_type compression,
                                     size_t max_total_uncomp_size)
 {
   size_t temp_size = 0;
-  auto const nvcomp_status =
+  auto nvcomp_status =
     batched_decompress_get_temp_size_ex(
-      compression, num_chunks, max_uncomp_chunk_size, &temp_size, max_total_uncomp_size)
-      .value_or(batched_decompress_get_temp_size(
```

Comment:
Are we inadvertently changing the semantics here?

The intention of the original code seemed to be to evaluate the "else" path *only* if `std::nullopt`. i.e. If `batched_decompress_get_temp_size_ex()` returned `nvcompErrorInternal`, it was *not* to call `batched_decompress_get_temp_size()`.

In the new version, `batched_decompress_get_temp_size()` is called in both cases. @abellina , @vuule, is that ok?

Reply:
It seems like what we should be checking is if nvcomp_status is simply nullopt, then doing the second call.

Reply:
It seems like what we should be checking is if nvcomp_status is simply nullopt, then doing the second call.

Reply:
Yeah - I'm not sure how much value there is in calling the second if the first failed in the library. It will likely also fail.

Reply:
I don't see a reason against trying the old API if the new one failed. Agreed that it's unlikely to help.